velox/expression/SimpleFunctionAdapter.h

```diff
@@ -80,7 +80,11 @@ class SimpleFunctionAdapter : public VectorFunction {
         EvalCtx* _context,
         VectorPtr* _result)
         : rows{_rows}, context{_context} {
-      BaseVector::ensureWritable(*rows, outputType, context->pool(), _result);
+      if (!*_result) {
+        *_result = context->getVector(outputType, _rows->size());
+      } else {
+        BaseVector::ensureWritable(*rows, outputType, context->pool(), _result);
```

Comment:
We only use the memory pool in ensureWritable to create vectors, if you were to pass in the VectorPool instead (which is available from the context as well), you could avoid this if statement, and vectors would get reused outside of just the SimpleFunctionAdapter.

This would be a big refactor, if you decide to do that, please don't do it in this diff.